src/components/ArticleVotes.jsx

```diff
@@ -0,0 +1,101 @@
+import { Box, Typography } from "@mui/material";
+import { useState, useEffect } from "react";
+import { upvoteArticle } from "../utils/patch";
+
+export default function ArticleVotes({ votes, articleId }) {
+  const [err, setErr] = useState("null");
+  const [articleVotes, setArticleVotes] = useState(0);
+  const [upvoteClicked, setUpvoteClicked] = useState(false);
+  const [downvoteClicked, setDownvoteClicked] = useState(false);
+  const [undoClicked, setUndoClicked] = useState(false);
+  useEffect(() => {
+    setArticleVotes(votes);
+  }, [votes]);
+
+  const handleClickUpvote = () => {
+    setArticleVotes(articleVotes + 1);
+    setErr(null);
+    setUpvoteClicked(true);
+    setUndoClicked(false);
+    upvoteArticle(articleId, { inc_votes: 1 }).catch((err) => {
+      setArticleVotes(articleVotes);
+      setUpvoteClicked(false);
+      setErr("Cannot change votes. Please try again");
+    });
+  };
+  const handleClickDownvote = () => {
+    setArticleVotes(articleVotes - 1);
+    setErr(null);
+    setDownvoteClicked(true);
+    setUndoClicked(false);
+    upvoteArticle(articleId, { inc_votes: -1 }).catch((err) => {
+      setArticleVotes(articleVotes);
+      setDownvoteClicked(false);
+      setErr("Cannot change votes. Please try again");
+    });
+  };
+
+  const handleClickUndoUpvote = () => {
+    setArticleVotes(articleVotes - 1);
+    setErr(null);
+    setUndoClicked(true);
+    setUpvoteClicked(false);
+    upvoteArticle(articleId, { inc_votes: -1 }).catch((err) => {
+      setArticleVotes(articleVotes);
+      setDownvoteClicked(false);
+      setErr("Cannot change votes. Please try again");
+    });
+  };
+  const handleClickUndoDownvote = () => {
+    setArticleVotes(articleVotes + 1);
+    setErr(null);
+    setUndoClicked(true);
+    setDownvoteClicked(false);
+    upvoteArticle(articleId, { inc_votes: -1 }).catch((err) => {
+      setArticleVotes(articleVotes);
+      setDownvoteClicked(false);
+      setErr("Cannot change votes. Please try again");
+    });
+  };
+
```

Comment:
Functionality works well and you've implemented optimistic rendering well! 🏆 
It also reacts well to errors happening and returns back to the previous state in a visual way the user can see 👍 
However a lot of repeated code here, you may want to consider trying to make a more flexible function that can be reused to impact state on the page (this isn't mandatory just a way we can improve this page)